infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationStreamNameTableManager.java

```diff
@@ -70,48 +79,71 @@ public class LogReplicationStreamNameTableManager {
                     TableRegistry.PROTOBUF_DESCRIPTOR_TABLE_NAME))
     ));
 
+    @Getter
+    private static String currentVersion;
+
+    private Table<TableInfo, Namespace, CommonTypes.Uuid> streamsNameTable;
+
+    private Table<VersionString, Version, CommonTypes.Uuid> pluginVersionTable;
+
+    @VisibleForTesting
+    public LogReplicationStreamNameTableManager(CorfuRuntime runtime) {
+        this.corfuStore = new CorfuStore(runtime);
+        this.pluginConfigFilePath = EMPTY_STR;
+    }
+
     public LogReplicationStreamNameTableManager(CorfuRuntime runtime, String pluginConfigFilePath) {
         this.pluginConfigFilePath = pluginConfigFilePath;
         this.corfuStore = new CorfuStore(runtime);
 
         initStreamNameFetcherPlugin();
+        openTables();
     }
 
-    public Set<String> getStreamsToReplicate() {
-        // Initialize the streamsToReplicate
-        if (verifyTableExists(LOG_REPLICATION_PLUGIN_VERSION_TABLE) &&
-            verifyTableExists(LOG_REPLICATION_STREAMS_NAME_TABLE)) {
-            // The tables exist but may have been created by another runtime in which case they have to be opened with
-            // key/value/metadata type info
-            openExistingTable(LOG_REPLICATION_PLUGIN_VERSION_TABLE);
-            openExistingTable(LOG_REPLICATION_STREAMS_NAME_TABLE);
-            if (!tableVersionMatchesPlugin()) {
-                // delete the tables and recreate them
-                deleteExistingStreamNameAndVersionTables();
-                createStreamNameAndVersionTables(
-                    logReplicationConfigAdapter.fetchStreamsToReplicate());
-            }
-        } else {
-            // If any 1 of the 2 tables does not exist, delete and recreate them both as they may have been corrupted.
-            deleteExistingStreamNameAndVersionTables();
-            createStreamNameAndVersionTables(
-                logReplicationConfigAdapter.fetchStreamsToReplicate());
+    private void openTables() {
+        try {
+            openTable(LOG_REPLICATION_PLUGIN_VERSION_TABLE);
+            openTable(LOG_REPLICATION_STREAMS_NAME_TABLE);
+        } catch (Exception e) {
+            log.error("Fatal error: Failed to open config tables", e);
+            throw new UnrecoverableCorfuError(e);
         }
-        return readStreamsToReplicateFromTable();
     }
 
-    public boolean isUpgraded() {
-        if (verifyTableExists(LOG_REPLICATION_PLUGIN_VERSION_TABLE)) {
-            openExistingTable(LOG_REPLICATION_PLUGIN_VERSION_TABLE);
-            return !tableVersionMatchesPlugin();
+    public Set<String> getStreamsToReplicate() {
+        try {
+            currentVersion = logReplicationConfigAdapter.getVersion();
+            Set<String> fetchedStreams = logReplicationConfigAdapter.fetchStreamsToReplicate();
+
+            IRetry.build(IntervalRetry.class, () -> {
```

Comment:
we need a 'return' here.  Take a look at RestoreRedundancyMergeSegments.performStateTransfer()